File: FormalBench-data/FilteredData/specifications/seed_726_per_category_10_653ade686f/java/NextPowerOf2.java
import java.io.*;
import java.lang.*;
import java.util.*;
import java.math.*;


class NextPowerOf2 {
    
    /*@ public normal_behavior
      @ assignable \nothing;
      @ diverges n > 1073741824;
      @ ensures n <= 1073741824;
      @ ensures 1 <= \result <= 1073741824 && \result >= n;
      @ ensures (\result & (\result - 1)) == 0;
      @ ensures \result == 1 || \result / 2 < n;
      @*/
    //@ code_java_math
    public static int nextPowerOf2(int n) {
        if (n == 0) {
            return 1;
        }
        int i = 1;
        /*@ loop_invariant i == 0 || i == Integer.MIN_VALUE || (1 <= i <= 1073741824 && (i & (i - 1)) == 0);
          @ loop_invariant n <= 1073741824 ==> 1 <= i <= 1073741824;
          @ loop_invariant n <= 1073741824 ==> i == 1 || i / 2 < n;
          @ loop_writes i;
          @*/
        while (i < n) {
            i = i << 1;
        }
        return i;
    }
}

File: FormalBench-data/FilteredData/specifications/seed_726_per_category_10_653ade686f/c/NextPowerOf2.c
#include "java_arrays.h"
#include <stdint.h>

/*@
  assigns \nothing;
  ensures \result == (int32_t)(2 * (integer)value);
*/
static int32_t java_shl_one(int32_t value) {
    return (int32_t)((uint32_t)value << 1);
}

/*@
  terminates n <= 1073741824;
  assigns \nothing;
  exits \false;
  ensures n <= 1073741824;
  ensures 1 <= \result <= 1073741824 && \result >= n;
  ensures (\result & (\result - 1)) == 0;
  ensures \result == 1 || \result / 2 < n;
*/
int32_t nextPowerOf2(int32_t n) {
    if (n == 0) {
        return 1;
    }
    int32_t i = 1;
    /*@
      loop invariant i == 0 || i == INT32_MIN || (1 <= i <= 1073741824 && (i & (i - 1)) == 0);
      loop invariant n <= 1073741824 ==> 1 <= i <= 1073741824;
      loop invariant n <= 1073741824 ==> i == 1 || i / 2 < n;
      loop assigns i;
    */
    while (i < n) {
        i = java_shl_one(i);
    }
    return i;
}
